src/sasjs-request/index.js

```diff
@@ -0,0 +1,103 @@
+import chalk from 'chalk'
+import path from 'path'
+import SASjs from '@sasjs/adapter/node'
+import { findTargetInConfiguration } from '../utils/config-utils'
+import { readFile } from '../utils/file-utils'
+import { getVariable } from '../utils/utils'
+import { getAccessToken } from '../utils/auth-utils'
+
+export async function runSasJob(
+  sasJobLocation,
+  dataFilePath,
+  configFilePath,
+  targetName
+) {
+  const { target, isLocal } = await findTargetInConfiguration(targetName)
+  if (!target) {
+    throw new Error('Target not found! Please try again with another target.')
+  }
+
+  let dataJson
+  let configJson
+
+  if (dataFilePath !== 'default') {
+    if (dataFilePath.split('.').slice(-1)[0] !== 'json') {
+      throw new Error('Provided data file must be valid json.')
+    }
+
+    const dataFile = await readFile(path.join(process.cwd(), dataFilePath))
+
+    try {
+      dataJson = JSON.parse(dataFile)
+    } catch (err) {
+      throw new Error('Provided data file must be valid json.')
+    }
+  }
+
+  if (configFilePath !== 'default') {
+    if (configFilePath.split('.').slice(-1)[0] !== 'json') {
+      throw new Error('Provided config file must be valid json.')
+    }
+
+    const configFile = await readFile(path.join(process.cwd(), configFilePath))
+
+    try {
+      configJson = JSON.parse(configFile)
+    } catch (err) {
+      throw new Error('Provided config file must be valid json.')
+    }
+  }
+
+  const sasjs = new SASjs({
+    serverUrl: target.serverUrl,
+    appLoc: target.appLoc,
+    serverType: target.serverType,
+    useComputeApi: true
+  })
+
+  const clientId = await getVariable('client', target)
+  const clientSecret = await getVariable('secret', target)
+
+  if (!clientId) {
+    throw new Error(
+      'A client ID is required for SAS Viya deployments.\n Please ensure that `client` is present in your build target configuration or in your .env file, and try again.\n'
+    )
+  }
+
+  if (!clientSecret) {
+    throw new Error(
+      'A client secret is required for SAS Viya deployments.\n Please ensure that `secret` is present in your build target configuration or in your .env file, and try again.\n'
+    )
+  }
+
+  const accessToken = await getAccessToken(
+    sasjs,
+    clientId,
+    clientSecret,
+    target,
+    isLocal
+  )
+
+  let data = {
+    fromjs: [dataJson]
+  }
+
+  await sasjs
+    .request(
+      sasJobLocation,
+      data,
+      configJson,
+      () => {
+        chalk.redBright(`Login callback called. Request failed.`)
+      },
+      accessToken
+    )
+    .then(
+      (res) => {
+        console.log('Request finish', res)
+      },
+      (err) => {
+        chalk.redBright(`Login callback called. Request failed. ${err}`)
```

Comment:
@YuryShkoda has added a `displayResult` function in the context command which contains some extra logic for displaying errors better. Maybe we could pull that out into its own file and use it here?

Reply:
I already moved displayResult() to a separate utility. Will be implemented in the upcoming PR.